src/clib/pio_rearrange.c

```diff
@@ -332,9 +332,16 @@ int create_mpi_datatypes(MPI_Datatype mpitype, int msgcnt,
         if (mcount[i] > 0)
         {
             int len = mcount[i] / blocksize;
-            int displace[len];
+            int *displace = NULL;
             LOG((3, "blocksize = %d i = %d mcount[%d] = %d len = %d", blocksize, i, i,
                  mcount[i], len));
+
+            if (len > 0)
+            {
+                if (!(displace = malloc(len * sizeof(int))))
```

Comment:
Do we need a calloc() here (Not entirely apparent that all entries are initialized)?

Reply:
Use calloc should be safe. No issues with malloc so far, though.